pkg/gadgets/trace/sni/tracer/tracer.go

```diff
@@ -68,7 +67,12 @@ func parseSNIEvent(sample []byte) (*types.Event, error) {
 		Event: eventtypes.Event{
 			Type: eventtypes.NORMAL,
 		},
-		Name: name,
+		Pid:       bpfEvent.Pid,
+		Tid:       bpfEvent.Tid,
+		MountNsID: bpfEvent.MountNsId,
```

Comment:
Having the mountns, we could use the usual enricher and simplify the `ConnectToContainerCollection`. I did some experiments: https://github.com/inspektor-gadget/inspektor-gadget/commit/21fbac142060089d7c5a159d1445a9a3e45a3660. And it seems to work:

```bash
$ ./kubectl-gadget trace sni -A
NODE                  NAMESPACE             POD                   CONTAINER             PID         TID        COMM       NAME
master                                                                                  690         180537     containerd registry-1.docker.io
master                                                                                  690         181966     containerd auth.docker.io
master                                                                                  0           0                     www.inspektor-gadge…
master                                                                                  0           0                     www.inspektor-gadge…
master                default               mypod                 mypod                 182123      182123     wget       www.inspektor-gadge…
```

BTW, I noticed two things:

1. Name column needs more "weight" to avoid being truncated.
2. Some events were not enriched with PID info.

Reply:
That's cool. Should I do that in the same PR?

> Name column needs more "weight" to avoid being truncated.

They already have `width:30` which should be enough for `www.inspektor-gadget.io`. I am not sure which value to choose.
I filed this issue for an alternative way: https://github.com/inspektor-gadget/inspektor-gadget/issues/1291

> Some events were not enriched with PID info. I suppose they were created before the tracer.

Correct. Because of this, we should not only rely on `MountNsID` but continue to have the current mechanism for namespace & pod.